BAT/BCEngineMCMC.h

```diff
@@ -272,34 +289,43 @@ class BCEngineMCMC
     const std::vector<std::vector<double> >& GetScaleFactors() const
     { return fMCMCProposalFunctionScaleFactor; }
 
+    /**
+     * @return states of all chains */
+    const std::vector<ChainState>& GetChainStates() const
+    { return fMCMCStates; }
+
+    /**
+     * @param c index of the Markov chain
+     * @return state of chains */
+    const ChainState& GetChainStates(unsigned c) const
+    { return fMCMCStates.at(c); }
+
     /**
      * @return current point of each Markov chain */
-    const std::vector<std::vector<double> >& Getx() const
-    { return fMCMCx; }
+    std::vector<std::vector<double> > Getx() const;
```

Comment:
Carefully check where Getx() is called? Changing from reference to value leads to calls like this producing undefined behavior

```c++
std::vector<std::vector<double> > v(Getx().begin(), Getx().end();
```

If we can remove it because we have `GetX(unsigned c)`, we should

Reply:
`Getx()` is never called inside any of our code. I'd be happy to remove it.